core/geometry/src/curve/internalContexts/geometry2d/UnboundedLine2d..ts

```diff
@@ -0,0 +1,229 @@
+  }
+  /**
+   * Return a new implicit line with its reference point given relative to newOrigin, and its perpendicular vector
+   *    normalized.
+   * @return undefined if perpendicular vector has zero length.
+   * @param newOrigin origin coordinates to subtract from existing origin.
+  */
+  public cloneNormalizedFromOrigin(newOrigin?: Point2d): UnboundedLine2dByPointAndNormal | undefined {
+    const unitNormal = this.normal.normalize();
+    if (unitNormal === undefined)
+      return undefined;
+    if (newOrigin !== undefined) {
+      return new UnboundedLine2dByPointAndNormal(
+        Point2d.create(this.point.x - newOrigin.x, this.point.y - newOrigin.y),
+        Vector2d.create(unitNormal.x, unitNormal.y));
+    } else {
+      return new UnboundedLine2dByPointAndNormal(this.point.clone(),
+        Vector2d.create(unitNormal.x, unitNormal.y));
+    }
+  }
+  /**
+   * Return a new implicit line with given origin and unit normal based on this.
+   * * WARNING: Beware of confusing related function cloneNormalizedFromOrigin, which has different origin
+   *    logic.
+   * @return undefined if perpendicular vector has zero length.
+  */
+  public cloneNormalizedWithOrigin(newOrigin?: Point2d): UnboundedLine2dByPointAndNormal | undefined {
+    const unitNormal = this.normal.normalize();
+    if (unitNormal === undefined)
+      return undefined;
+    if (newOrigin === undefined)
+        return new UnboundedLine2dByPointAndNormal (this.point, unitNormal);
+    return new UnboundedLine2dByPointAndNormal(
+      Point2d.create(newOrigin.x, newOrigin.y),
+      Vector2d.create(unitNormal.x, unitNormal.y));
+  }
+
+  /**
+   * Return a new implicit line with its reference point shifted by given multiple of its normal vector.
+   * * The shift is applied to the point, and the normal copied unchanged-- no normalization or test for zero.
+   * @param shiftFactor multiplier for normal.
+   * @return shifted line
+  */
+  public cloneShifted(shiftFactor: number): UnboundedLine2dByPointAndNormal | undefined {
+    return UnboundedLine2dByPointAndNormal.createPointNormal(
+      this.point.plusScaled(this.normal, shiftFactor),
+      this.normal);
+  }
+
+  /**
+   * Compute the intersection of two lines.
+   * * Each can be offset.
+   * * Offsets are in units of the dot products with normal vectors.
+   *   * If normal vectors are unit vetors, the offsets are simple distance.
+   *   * Otherwise the offset scaling is scaled --- use carefully.
+   * @param other the other line.
+   * @param thisOffset target function value for dot proucts with this line
+   * @param otherOffset target function value for dot products with other line
+   */
+  public interesectUnboundedLine2dByPointAndNormalWithOffsets(other: UnboundedLine2dByPointAndNormal, thisOffset: number = 0, otherOffset: number = 0): Point2d | undefined {
+    const xy = Vector2d.create();
+    if (SmallSystem.linearSystem2d(
```

Comment:
Please add this comment to clarify

// points `p = (x,y)` on the line satisfies equation `normal dot (p−point​)=0`
// therefore:
// `normal.x ​(x−point.x​) + normal.y ​(y−point.y​) = thisOffset` or
// `normal.x ​x + normal.y ​y= thisOffset + normal dot point.x​`